core/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -1866,4 +1871,28 @@ public void testWithRandomException() throws IOException {
             }
         }
     }
+
+    public void testCheckpointOnDiskFull() throws IOException {
+        Checkpoint checkpoint = new Checkpoint(randomLong(), randomInt(), randomLong());
+        Path tempDir = createTempDir();
+        Checkpoint.write(FileChannel::open, tempDir.resolve("foo.cpk"), checkpoint, StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW);
+        Checkpoint checkpoint2 = new Checkpoint(randomLong(), randomInt(), randomLong());
+
+        try {
+            Checkpoint.write((p, o) -> {
+                FileChannel open = FileChannel.open(p, o);
```

Comment:
do we also want to sometime fail before we open the file? I get that will not make a file linger around but it will be good to so nothing else fails, right?